src/toy/blast/Blast.sol

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: BSL 1.1 - Copyright 2024 MetaLayer Labs Ltd.
+pragma solidity ^0.8.15;
+
+import {GasMode, IGas} from "./Gas.sol";
+
+enum YieldMode {
+  AUTOMATIC,
+  VOID,
+  CLAIMABLE
+}
+
+interface IYield {
+  function configure(address contractAddress, uint8 flags) external returns (uint);
+  function claim(address contractAddress, address recipientOfYield, uint desiredAmount) external returns (uint);
+  function getClaimableAmount(address contractAddress) external view returns (uint);
+  function getConfiguration(address contractAddress) external view returns (uint8);
+}
+
+contract Blast {
```

Comment:
Maybe this should this implement the `IBlast` interface to ensure they're in sync?